XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs

```diff
@@ -215,6 +260,12 @@ private void OnGamepadAdded(object sender, Gamepad e)
             _isGamePadConnected = Gamepad.Gamepads.Any();
         }
 
+        private void OnNavigationViewControlLoaded(object sender, RoutedEventArgs e)
+        {
+            // Delay necessary to ensure NavigationView visual state can match navigation
+            Task.Delay(500).ContinueWith(_ => this.NavigationViewLoaded?.Invoke(), TaskScheduler.FromCurrentSynchronizationContext());

```

Comment:
I'm not sure if this is the best way to ensure this.

Reply:
This is needed in conjunction with the change above to prevent a bad visual state where a group is expanded, but no item in the group has the selection visual. There are some reports of NavigationViewItem selection not always working the first time, which is what I was experiencing when testing protocol activation cold launches. It seems to be timing-related, and possibly impacted by the animation to expand the group.

Alternative solutions here might be to explore delaying protocol activation handling or possibly NavigationViewItem creation, but both of those would have an impact on launch perf. This one-time delayed call to EnsureNavigationSelection fixes the visual timing issue on cold launch without impacting launch perf.

Reply:
Oof I see. Is this purely related to a timing issue or possible a bug with NavigationView (namely https://github.com/microsoft/microsoft-ui-xaml/issues/3149 or https://github.com/microsoft/microsoft-ui-xaml/issues/3015)? If the latter, can you add a comment indicating that or possible create an issue to remove this once it's fixed in WinUI? That would be great!